roleutils/roles.py

```diff
@@ -214,6 +215,36 @@ async def role_color(
         await ctx.send(
             f"**{role}** color changed to **{color}**.", embed=await self.get_info(role)
         )
+        
+    @commands.admin_or_permissions(manage_roles=True)
+    @commands.bot_has_permissions(manage_roles=True)
+    @role.command("icon")
+    async def role_icon(
+        self,
+        ctx: commands.Conext,
+        role: StrictRole(check_integrated=False),
+        icon: Union[discord.Emoji, discord.PartialEmoji, str] = None,
+    ):
+        """Change a role's icon."""
+        if "ROLE_ICONS" not in ctx.guild.features:
+            return await ctx.send(
+                "Server does not have support for the role icons feature yet."
+            )
+            
+        if isinstance(icon, discord.Emoji):
+            await role.edit(display_icon=await icon.read())
+        elif isinstance(icon, discord.PartialEmoji) and icon.is_custom_emoji():
+            await role.edit(display_icon=await icon.read())
+        elif isinstance(icon, str) and icon.startswith("http"):
```

Comment:
I don't want to be accepting any arbitrary URL to make a request to as this could lead to security issues. Rather than fetching content from the image itself, I'd do these two things:
- check `message.attachments` and see if there's an image attachment that could be set as an icon
- check `message.embeds` and see if there's an embedded image that could be set as an icon

Reply:
Alright, understandable, I'm out for the week, will be done when i'm back home and have a stable internet.